src/js/bootstrap-switch.js

```diff
@@ -394,22 +394,38 @@ class BootstrapSwitch {
       this.setPrevOptions()
       this._width()
       this._containerPosition()
+
+      // Allow browser repaint to end of event queue but minimize flicker
       setTimeout(() => {
+        // Ensure visible in case hidden while calculating widths
+        this.$wrapper.css('visibility', 'visible')
         if (this.options.animate) {
           return this.$wrapper.addClass(this._getClass('animate'))
         }
-      }, 50)
+      }, 0)
     }
+
+    // initialize if the $wrapper is already visible
     if (this.$wrapper.is(':visible')) {
       init()
       return
     }
-    const initInterval = window.setInterval(() => {
+
+    let interval = 0.5
```

Comment:
I don't think you need to clear the timeout because it's already finished by the time the next one is called